src/go/tracing/tracing_test.go

```diff
@@ -340,16 +223,114 @@ func TestGCPTracingConfig(t *testing.T) {
 				if got == nil {
 					t.Errorf("Test (%s): failed, expected result should not be nil", tc.desc)
 				}
-				if got.Http.Name != "envoy.tracers.opencensus" {
-					t.Errorf("Test (%s): failed, expected config name is wrong", tc.desc)
+
+				if !proto.Equal(got, tc.wantResult) {
+					t.Errorf("Test (%s): failed, got : %v, want: %v", tc.desc, got, tc.wantResult)
 				}
+			}
+
+		})
+
+	}
+}
+
+// Tests the sample rate is correctly populated in the HCM tracing config.
+func TestHcmTracingSampleRate(t *testing.T) {
+
+	testData := []struct {
+		desc              string
+		tracingSampleRate float64
+		wantResult        *hcmpb.HttpConnectionManager_Tracing
+		wantError         string
+	}{
+		{
+			desc:              "Default sampling rate works",
+			tracingSampleRate: options.DefaultCommonOptions().TracingSamplingRate,
```

Comment:
do we have tests for disabled,  rate=0 and ratte=1?  If no, please add them

Reply:
Sure, added. There's nothing special about 0 and 1 anymore, it gets translated directly to 0% or 100%.

Reply:
And there are integration tests already for rate 0 and 1.